lib/codecov.js

```diff
@@ -474,15 +480,26 @@ var upload = function(args, on_success, on_failure) {
   } else if ((args.options.disable || '').split(',').indexOf('search') === -1) {
     console.log('==> Scanning for reports')
     var _files
+    var _findArgs
     if (!isWindows) {
-      _files = execSync('find ' + root + ' ' + patterns + more_patterns)
+      // @TODO support for a root directory
+      // It's not straightforward due to the nature of the find command
+      _findArgs = [root].concat(patterns)
+      if (more_patterns) {
+        _findArgs.concat(more_patterns)
+      }
+      _files = execFileSync('find', _findArgs)
         .toString()
         .trim()
         .split('\n')
     } else {
       // @TODO support for a root directory
       // It's not straightforward due to the nature of the dir command
-      _files = execSync('dir ' + patterns + more_patterns)
+      _findArgs = [root].concat(winPatterns)
+      if (more_patterns) {
+        _findArgs.concat(more_patterns)
+      }
+      _files = execSync('dir ' + winPatterns + more_patterns)
```

Comment:
```suggestion
      _files = execSync('dir ' + winPatterns.join(' ') + more_patterns)
```

`winPattern` is an array, using it like this inside of a string concatenation will automatically merge all the elements of the array as if called with `join(', ')`, which we do not want since `,` is not a valid char after a pipe operator. Converting to string before the concatenation using join should fix the issue.